.github/workflows/cd.yml

```diff
@@ -46,89 +39,64 @@ jobs:
-      # - name: Create Zcash incoming traffic firewall rule
-      #   run: |
-      #     gcloud compute firewall-rules create "allow-zcash" \
-      #     --target-tags zebrad \
-      #     --allow tcp:8233,tcp:18233 \
-      #     --source-ranges 0.0.0.0/0 \
-      #     --description="Allow incoming Zcash traffic from anywhere" \
-
       # Create instance template from container image
       - name: Create instance template
         run: |
-          gcloud compute instance-templates create-with-container "zebrad-$BRANCH_NAME-$SHORT_SHA" \
-          --container-image "gcr.io/$PROJECT_ID/$REPOSITORY/$BRANCH_NAME:$SHORT_SHA" \
-          --create-disk name=zebrad-cache-$SHORT_SHA,auto-delete=yes,size=100GB,type=pd-balanced \
-          --container-mount-disk mount-path="/zebrad-cache",name=zebrad-cache-$SHORT_SHA \
-          --machine-type n2d-standard-4 \
-          --service-account cos-vm@zealous-zebra.iam.gserviceaccount.com \
+          gcloud compute instance-templates create-with-container "zebrad-${{ env.GITHUB_HEAD_REF_SLUG || env.GITHUB_REF_SLUG }}-${{ env.GITHUB_SHA_SHORT }}" \
+          --container-image "gcr.io/${{ env.PROJECT_ID}}/${{ env.GITHUB_REPOSITORY_NAME_PART_SLUG }}/${{ env.GITHUB_HEAD_REF_SLUG || env.GITHUB_REF_SLUG }}:${{ env.GITHUB_SHA_SHORT }}" \
+          --create-disk name=zebrad-cache-${{ env.GITHUB_SHA_SHORT }},auto-delete=yes,size=100GB,type=pd-balanced \
+          --container-mount-disk mount-path="/zebrad-cache",name=zebrad-cache-${{ env.GITHUB_SHA_SHORT }} \
+          --machine-type ${{ env.MACHINE_TYPE }} \
+          --service-account ${{ env.DEPLOY_SA }} \
           --scopes cloud-platform \
           --tags zebrad \
 
-      # Run once: create firewall rule to allow healthchecks
-      # - name: Create healthcheck firewall rule
-      #   run: |
-      #     gcloud compute firewall-rules create "allow-tracing-health-check" \
-      #     --target-tags zebrad \
-      #     --allow tcp:3000 \
-      #     --source-ranges 130.211.0.0/22,35.191.0.0/16 \
-      #     --description="Allow HTTP requests to our tracing endpoint from Google's probes" \
-
       # Check if our destination instance group exists already
       - name: Check if instance group exists
         id: does-group-exist
         continue-on-error: true
         run: |
-          gcloud compute instance-groups list | grep "zebrad-$BRANCH_NAME" | grep "$REGION"
+          gcloud compute instance-groups list | grep "zebrad-${{ env.GITHUB_HEAD_REF_SLUG || env.GITHUB_REF_SLUG }}" | grep "${{ env.REGION }}"
 
       # Deploy new managed instance group using the new instance template
       - name: Create managed instance group
         if: steps.does-group-exist.outcome == 'failure'
         run: |
           gcloud compute instance-groups managed create \
-          "zebrad-$BRANCH_NAME" \
-          --template "zebrad-$BRANCH_NAME-$SHORT_SHA" \
+          "zebrad-${{ env.GITHUB_HEAD_REF_SLUG || env.GITHUB_REF_SLUG }}" \
```

Comment:
Can this be set once and reused by the rest of the workflow?

Reply:
That would require a bit more refactoring in this workflow. I'm working on another branch on the refactor, so we'll have a PR with this recommendation soon enough.